test/manager.test.ts

```diff
@@ -0,0 +1,855 @@
+     */
+    testSNApp.componentManager.deactivateComponent(testThemeDefault.uuid);
+
+    /**
+     * componentManager.postActiveThemesToComponent() will trigger the ActivateTheme action.
+     * This should deactivate the Default theme, and activate the Dark theme.
+     */
+    testSNApp.componentManager.postActiveThemesToComponent(testComponent);
+    await sleep(0.001);
+
+    const customThemes = childWindow.document.head.getElementsByClassName('custom-theme');
+    expect(customThemes.length).toEqual(1);
+
+    const themeLink = customThemes[0] as HTMLLinkElement;
+    expect(themeLink.id).toEqual(btoa(testThemeDark.hosted_url));
+    expect(themeLink.href).toEqual(new URL(testThemeDark.hosted_url).href);
+    expect(themeLink.type).toEqual('text/css');
+    expect(themeLink.rel).toEqual('stylesheet');
+    expect(themeLink.media).toEqual('screen,print');
+  });
+
+  it('should queue message if sessionKey is not set', async () => {
+    /**
+     * Messages are queued when the sessionKey is not set or has a falsey value.
+     * sessionKey is set by Uuid.GenerateUuid() which uses our generateUuid
+     * function in our Utils module. We will mock the return value to be undefined.
+     */
+    const Utils = require('./../lib/utils');
+    const originalGenerateUuid = Utils.generateUuid;
+    Utils.generateUuid = jest.fn().mockReturnValue(undefined);
+
+    await registerComponent(testSNApp, childWindow, testComponent);
+    const parentPostMessage = jest.spyOn(childWindow.parent, 'postMessage');
+    componentManager.setComponentDataValueForKey("testing", "1234");
+    expect(parentPostMessage).not.toHaveBeenCalled();
+
+    // Restoring the mocked function.
+    Utils.generateUuid = originalGenerateUuid;
+  });
+
+  describe('component actions', () => {
+    beforeAll(() => {
+      /**
+       * Components prompts for permissions via the presentPermissionsDialog function, which
+       * has been implemented to use window.confirm to approve these requests.
+       * We want to approve all of them during our ComponentActions tests.
+       */
+      window.confirm = (message) => true;
+    });
+
+    afterAll(() => {
+      // Restoring window.confirm implementation.
+      window.confirm = (message) => false;
+    });
+
+    test('streamItems', async (done) => {
+      expect.hasAssertions();
+  
+      const savedTestNote = await createNoteItem(testSNApp);
+      const contentTypes = [
+        ContentType.Note
+      ];
+
+      await registerComponent(testSNApp, childWindow, testComponent);
+  
+      componentManager.streamItems(contentTypes, (items) => {
+        expect(items).not.toBeUndefined();
+        expect(items.length).toBeGreaterThanOrEqual(0);
+        expect(items[0].uuid).toBe(savedTestNote.uuid);
+        done();
+      });
+    });
+  
+    test('streamContextItem', async () => {
+      expect.assertions(8);
+  
+      const simpleNote = await createNoteItem(testSNApp, {
+        title: 'A simple note',
```

Comment:
Earlier this year we've found an encoding bug in sncrypto, so now we try to include characters that fall outside of the ascii range in our dummy strings (emojis/non-roman characters are great candidates for this)